[ReactRoles] Add support for button roles

Evanroby opened the pull request:
In the recent commits I've made, I updated the work of reaction roles, by adding a new file `buttonroles.py`, which is exactly same as reactroles, but using buttons.

Evanroby committed:
[ButtonRoles] Add button role support with DynamicItem

Evanroby committed:
[RoleUtils] added compatibility with buttonroles

Evanroby committed:
[ButtonRoles] added support for buttonroles

japandotorg commented on roleutils/buttonroles.py:
```diff
@@ -0,0 +1,441 @@
+        }
+        self.config.register_guild(**default_guild)
+
+    async def cog_load(self) -> None:
+        """Start async task after Red is fully ready and config is registered."""
+        self._task = asyncio.create_task(self._initialize_button_roles())
+
+    async def cog_unload(self) -> None:
+        """Clean up tasks when cog unloads."""
+        if self._task:
+            self._task.cancel()
```
You're duplicating the `cog_load` & `cog_unload` methods here, use `initialize` instead.

japandotorg commented on roleutils/buttonroles.py:
```diff
@@ -0,0 +1,441 @@
+        self.cache["buttonroles"] = {"message_cache": set()}
+        self._task: Optional[asyncio.Task] = None
+
+        default_guild = {
+            "buttonroles": {}
+        }
+        self.config.register_guild(**default_guild)
```
this is a duplicate of `L99`  in the `RoleUtils` file in this PR, remove this

japandotorg commented on roleutils/buttonroles.py:
```diff
@@ -0,0 +1,441 @@
+
+    def __init__(self, *_args: Any) -> None:
+        super().__init__(*_args)
+        self.method: str = "build"
```
~~no need to use this here, remove~~ (canceling this one rn, you're somehow making use of it, we'll try to adjust that later)

japandotorg commented on roleutils/buttonroles.py:
```diff
@@ -0,0 +1,441 @@
+
+    async def cog_load(self) -> None:
+        """Start async task after Red is fully ready and config is registered."""
+        self._task = asyncio.create_task(self._initialize_button_roles())
```
also use `self.create_task` instead of `asyncio.create_task` in initialize

japandotorg reviewed (changes requested):
requested changes must be fixed/implemented, review will only continue after that again.

Evanroby committed:
[ButtonRoles]: simplify initialization

- Removed duplicate config registration
- Replaced cog_load/unload with initialize
- Used self.create_task instead of asyncio.create_task
- Renamed _update_cache to avoid conflict

Evanroby committed:
[ButtonRoles]: simplify initialization

- Removed duplicate config registration
- Replaced cog_load/unload with initialize
- Used self.create_task instead of asyncio.create_task
- Renamed _update_cache to avoid conflict
- Added back license, didn't copy it before

Evanroby commented:
just did commits! feel free to look into it!

Evanroby committed:
[ButtonRoles]: simplify initialization

- Removed duplicate config registration
- Replaced cog_load/unload with initialize (totally forgot to do it earlier, but logged it, sorry)
- Used self.create_task instead of asyncio.create_task
- Renamed _update_cache to avoid conflict
- Added back license, didn't copy it before

Evanroby commented:
totally had forgot to do the `cog_load` things, sorry im in rush because I got school in a small amount of time

Evanroby committed:
[ButtonRoles]: Fixed self.method conflicting with reactroles.py

- Renamed `self.method` to `self.button_method`

Evanroby committed:
Update buttonroles.py

Fixed self.button conflicting with reactroles

Evanroby committed:
WIP: local changes before pulling

Evanroby committed:
WIP: local changes before pulling

Evanroby committed:
[ButtonRoles]: Deleted database usage

Removed all Config database usage
Replaced with in-memory active_button_roles dictionary

japandotorg committed:
[ExtendedMentionHelp] add missing req to `info.json`

japandotorg committed:
[Applications] fixes and event role triggers

japandotorg committed:
Merge branch 'main' of https://github.com/japandotorg/Seina-Cogs

japandotorg committed:
Style Reformatting

japandotorg committed:
[Applications] fix command params for event roles

japandotorg committed:
Style Reformatting

japandotorg committed:
[Applications] fixes, adjustments and a couple more new features

japandotorg committed:
Style Reformatting

japandotorg committed:
[Applications] add a hidden "backup" command for as long as the cog is in BETA

japandotorg committed:
Style Reformatting

japandotorg committed:
[Applications] add a voters system to responses

japandotorg committed:
Style Reformatting

japandotorg committed:
[Applications] fix role assigment

japandotorg committed:
[Applications] fix dynamic voter buttons custom_id

japandotorg committed:
Style Reformatting

japandotorg committed:
[Applications] temp fix for role assigments

japandotorg committed:
Style Reformatting

japandotorg committed:
[Applications] fixed role assignments finally

Evanroby committed:
Fix for buttonroles

Evanroby committed:
Merge branch 'main' of https://github.com/Evanroby/Seina-Cogs

Evanroby committed:
[Reactroles] self.method, not self.button_method
